test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -587,3 +590,89 @@ def test_query_workspace_with_load_id_mass(setup_and_teardown):
+            [2, 'assembly_2', 'KBaseGenomeAnnotations.Assembly-10', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {}]
+        ]
+    }
+
+    genome_objs_response = {
+        "infos": [
+            [1, 'genome_1', 'KBaseGenomes.Genome-6', 'time', 75, 'user', 42, 'wsname', 'md5', 78,
+             {"load_id": load_id, "Assembly Object": "1/1/1"}],
+            [4, 'genome_2', 'KBaseGenomes.Genome-9.3', 'time', 75, 'user', 42, 'wsname', 'md5', 78,
+             {"load_id": load_id, "Assembly Object": "2/2"}]
+        ]
+    }
+
+    ws.get_object_info3.side_effect = [genome_objs_response, assembly_objs_response]
+
+    obj_names, obj_upas, assembly_objs_info, genome_objs_info = workspace_uploader._query_workspace_with_load_id_mass(
+        ws,
+        69046,
+        load_id,
+        ["genome_1", "genome_2",],
+        assembly_objs_only=False
+    )
+    assert obj_names == ["genome_1", "genome_2",]
+    assert obj_upas == ['42_1_75', '42_4_75']
+    assert assembly_objs_info == assembly_objs_response['infos']
+    assert genome_objs_info == genome_objs_response['infos']
+
+    # Assert expected calls to ws.get_object_info3
+    expected_calls = [
+        call({"objects": [{"wsid": 69046, "name": "genome_1"},
+                          {"wsid": 69046, "name": "genome_2"}], "ignoreErrors": 1, "includeMetadata": 1}),
+        call({"objects": [{"ref": "1/1/1"}, {"ref": "2/2"}], "includeMetadata": 1}),
+    ]
+    ws.get_object_info3.assert_has_calls(expected_calls, any_order=False)
+
+    # Test case 2: Invalid scenario - genome object does not have an 'Assembly Object' field in its metadata
+    genome_objs_response = {
+        "infos": [
+            [1, 'genome_1', 'KBaseGenomes.Genome-6', 'time', 75, 'user', 42, 'wsname', 'md5', 78,
+             {"load_id": load_id}]
+        ]
+    }
+
+    ws.get_object_info3.side_effect = [genome_objs_response]
+    with pytest.raises(ValueError) as excinfo:
+        workspace_uploader._query_workspace_with_load_id_mass(
+            ws,
+            69046,
+            load_id,
+            ["genome_1"],
+            assembly_objs_only=False
+        )
+    assert "Genome objects must have an 'Assembly Object' field in their metadata" == str(excinfo.value)
+
+
+def test_check_obj_type():
+    # Test case 1: Valid scenario
+    workspace_id = 1
+    load_id = "12345"
+    obj_infos = [
+        [1, 'abc-123', 'TypeA', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {'foo': 'bar'}],
+        [2, 'def-456', 'TypeB', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {}]
+    ]
+    expected_obj_types = {'TypeA', 'TypeB'}
+
+    workspace_uploader._check_obj_type(workspace_id, load_id, obj_infos, expected_obj_types)
+
+    # Test case 2: Invalid scenario
+    obj_infos = [
+        [1, 'abc-123', 'TypeA', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {'foo': 'bar'}],
+        [2, 'def-456', 'TypeC', 'time', 75, 'user', 42, 'wsname', 'md5', 78, {}]  # TypeC is not expected
+    ]
```

Comment:
Less code to just change expected_obj_types, unless there's a reason not to do that

Reply:
👍

Reply:
:+1: